sds/src/scanner/mod.rs

```diff
@@ -2559,6 +2563,44 @@ mod test {
         assert_eq!(matches[1].match_status, MatchStatus::Valid);
         assert_eq!(matches[2].match_status, MatchStatus::Valid);
     }
+
+    #[tokio::test]
+    async fn test_mock_endpoint_with_multiple_hosts() {
```

Comment:
I think we also want to tests a few other cases. In particular, it would be interesting to validate that:
- if all status are invalid, then the result is invalid. 
- if one status is valid and one is erroring, then the result is valid
- If one status is erroring and all others are invalid, then the result is erroring (unchecked), [because we don't know if the erroring status validate the match or invalidate it]

Reply:
this is already tested in the match_status.rs via the merge function.
I don't think we should cover all cases here the idea is to confirm that the merge is called. 
merge is already unit tested